### PR TITLE
Unify fleet and stand-alone suites

### PR DESCRIPTION
**Pull request by jalvz**
## What does this PR do?

Unifies fleet and stand alone suites

## Why is it important?

This makes steps reusable 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the Unit tests for the CLI, and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

Closes https://github.com/elastic/e2e-testing/issues/1109

### CODE REVIEW

File: e2e/_suites/fleet/fleet.go [hunk 1]
```diff
@@ -160,6 +185,219 @@ func (fts *FleetTestSuite) contributeSteps(s *godog.ScenarioContext) {
...
+	s.Step(`^a "([^"]*)" stand-alone agent is deployed with fleet server mode on cloud$`, fts.aStandaloneAgentIsDeployedWithFleetServerModeOnCloud)
+	s.Step(`^there is new data in the index from agent$`, fts.thereIsNewDataInTheIndexFromAgent)
+	s.Step(`^the "([^"]*)" docker container is stopped$`, fts.theDockerContainerIsStopped)
+	s.Step(`^there is no new data in the index after agent shuts down$`, fts.thereIsNoNewDataInTheIndexAfterAgentShutsDown)
+	s.Step(`^the stand-alone agent is listed in Fleet as "([^"]*)"$`, fts.theStandaloneAgentIsListedInFleetWithStatus)
+	s.Step(`^the "([^"]*)" integration is added to the policy$`, fts.theIntegrationIsAddedToThePolicy)
+	s.Step(`^the "([^"]*)" datasource is shown in the policy$`, fts.thePolicyShowsTheDatasourceAdded)
+}
+
+func (fts *FleetTestSuite) theIntegrationIsAddedToThePolicy(packageName string) error {
```
Comment (mdelapenya) [inline e2e/_suites/fleet/fleet.go:201]: We can still have standalone.go file with these functions right? To physically separate concerns, although hanging from the Fleet test suite struct umbrella

File: e2e/_suites/fleet/stand-alone.go [hunk 2]
```diff
@@ -177,35 +145,52 @@ func (sats *StandAloneTestSuite) startAgent(image string, composeFilename string
...
 }
 
-func (sats *StandAloneTestSuite) getContainerLogs() error {
-	serviceManager := compose.NewServiceManager()
+func (fts *FleetTestSuite) theIntegrationIsAddedToThePolicy(packageName string) error {
+	return theIntegrationIsOperatedInThePolicy(fts.kibanaClient, fts.FleetServerPolicy, packageName, "added")
+}
+
+func (fts *FleetTestSuite) thePolicyShowsTheDatasourceAdded(packageName string) error {
+	fts.StandAlone = true
```
Comment (mdelapenya) [inline e2e/_suites/fleet/stand-alone.go:164]: Is this line required? I can see two different steps calling `thePolicyShowsTheDatasourceAdded`, one for standalone and one for "regular" fleet enrollments. See `fleet.go:L164` and `fleet.go:L181`. We could reuse the existing Gherkin step, or if we need to distinguish both modes, we could create another function calling the above one, but setting `.Standalone = true` before it.

Comment (jalvz) [reply e2e/_suites/fleet/stand-alone.go]: right, that was a leftover - no needed. fixed

File: e2e/_suites/fleet/stand-alone.go [hunk 3]
```diff
@@ -177,35 +142,47 @@ func (sats *StandAloneTestSuite) startAgent(image string, composeFilename string
...
 
-func (sats *StandAloneTestSuite) getContainerLogs() error {
-	serviceManager := compose.NewServiceManager()
+func (fts *FleetTestSuite) thePolicyShowsTheDatasourceAdded(packageName string) error {
+	log.WithFields(log.Fields{
+		"policyID": fts.FleetServerPolicy.ID,
+		"package":  packageName,
+	}).Trace("Checking if the policy shows the package added")
+
+	maxTimeout := time.Minute
```
Comment (mdelapenya) [inline e2e/_suites/fleet/stand-alone.go:162]: Wdyt about using `common.TimeoutFactor` here as a multiplier (default value is 5 on CI, but configurable in Jenkins UI)? If the 1 minute is enough then I'm OK with this change.

We use the factor for CI, where sometimes we want to verify if waiting more for a condition allows the test code to reach the desired state

File: e2e/_suites/fleet/world.go [hunk 4]
```diff
@@ -13,16 +13,15 @@ import (
 
 // IngestManagerTestSuite represents a test suite, holding references to the pieces needed to run the tests
 type IngestManagerTestSuite struct {
-	Fleet      *FleetTestSuite
-	StandAlone *StandAloneTestSuite
+	Fleet *FleetTestSuite
```
Comment (mdelapenya) [inline e2e/_suites/fleet/world.go:16]: For follow-up, we can simply merge everything into the `IngestManagerTestSuite`, but I'm totally OK doing that in a follow-up PR.

**Review by mdelapenya — APPROVED** [review]
I added a few minor comments, not blockers at all, so LGTM.

We can discuss them here or in follow-up PRs.

Thank you for simplifying stuff :)

File: e2e/_suites/fleet/stand-alone.go (hunk 3 above)
Comment (jalvz) [reply e2e/_suites/fleet/stand-alone.go:162]: Sure, it was that done way - this PR simply copies  the function, no code changes.

**Pull request merged**